### ISSUE TITLE
Polling nvme drive temperature too regularly actually causes high temperatures

### ISSUE THREAD

**Issue by RoverAz** [lm-sensors/lm-sensors#511]
Not sure whether to file this under software issue, hardware quirk or user-is-dumb-and-should-know-better.

In short: regularly polling my nvme drive temperature with lm-sensors or anything that uses it will actually cause unusual heating in the drive. Polling once a second (which was the default for the lm-sensors-using kde widget I was using) will cause, in a matter of minutes, my nvme drive to reach and idle at ~50C when it is under no unusual load. No sensors and it sits at a happy 30C. I turned off the kde widget and managed to reproduce it with `watch -n1 "sensors | grep nvme -A4"`

```
... start ...
nvme-pci-0200
Adapter: PCI adapter
Composite:    +30.9°C  (low  = -60.1°C, high = +89.8°C)
                       (crit = +94.8°C)
... 1 min later ...
nvme-pci-0200
Adapter: PCI adapter
Composite:    +36.9°C  (low  = -60.1°C, high = +89.8°C)
                       (crit = +94.8°C)
... 2 mins later ...
nvme-pci-0200
Adapter: PCI adapter
Composite:    +41.9°C  (low  = -60.1°C, high = +89.8°C)
                       (crit = +94.8°C)
```

At about ~10 mins it will hit ~50C where it will stay until I stop the sensors.

I'm using a Gigayte G5 KF laptop
`02:00.0 Non-Volatile memory controller: Phison Electronics Corporation E16 PCIe4 NVMe Controller (rev 01)`

But... I'm wondering...
There are many posts online with people worrying about their nvme drives idling at 50C, could this be why? ~~Schroedinger's overheating drive~~ Overeager temperature measurement itself causing the heating? Is it just my hardware or does this happen with other nvme drives?

**Comment by Wer-Wolf** [comment]
It could be that reading the temperature causes the nvme drive to leave some sort of powersaving mode, leading to higher power consumtion.

Maybe a firmware update is available for the nvme driver?

**Comment by haarp** [comment]
Yes, likely forcing the NVMe to leave power-saving states. This is also a problem with the `hddtemp` driver which will spin up sleeping HDDs just to poll hte temperature. lm-sensors needs a mode that prefers showing no data over waking devices.

**Comment by Wer-Wolf** [comment]
That will be difficult to implement, as not all nvme drives suffer from this problem.

I suggest that you reduce the polling interval and check for firmware updates.
